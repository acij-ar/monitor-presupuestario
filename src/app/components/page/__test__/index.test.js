/**
 * @jest-environment jsdom
 */

const React = require('react');
const { shallow } = require('enzyme');

const mockPageview = jest.fn();
jest.mock('../../analytics', () => ({ pageview: mockPageview }));

const Page = require('..');

describe('Page component', () => {
  beforeEach(() => {
    mockPageview.mockClear();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Page pageName="mocked-page" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the analytics pageview method', () => {
    Object.defineProperty(window, 'location', { value: { pathname: 'mocked-url'} });
    shallow(<Page pageName="mocked-page" />);
    expect(mockPageview).toHaveBeenCalledTimes(1);
    expect(mockPageview).toHaveBeenCalledWith('mocked-url');
  });
});
