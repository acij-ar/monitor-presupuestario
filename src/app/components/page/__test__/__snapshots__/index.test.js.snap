// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component should match snapshot 1`] = `
<Fragment>
  <div
    id="main"
    key="main"
  >
    <Menu
      selectedPage="mocked-page"
    />
    <div
      id="page-content"
    />
  </div>
  <Footer
    key="footer"
  />
</Fragment>
`;
